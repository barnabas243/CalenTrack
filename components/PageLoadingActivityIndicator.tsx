import React from 'react';
import {ActivityIndicator, View, StyleSheet} from 'react-native';
import {useTheme} from 'react-native-paper';

const PageLoadingActivityIndicator = () => {
  const {colors} = useTheme();
  return (
    <View
      testID="activity-indicator-container"
      style={[styles.container, {backgroundColor: colors.background}]}>
      <ActivityIndicator testID="activity-indicator" size="large" color={colors.primary} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default PageLoadingActivityIndicator;
