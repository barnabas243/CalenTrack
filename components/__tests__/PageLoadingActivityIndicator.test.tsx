import React from 'react';
import {render} from '@testing-library/react-native';
import {ThemeProvider, DefaultTheme} from 'react-native-paper';
import PageLoadingActivityIndicator from '../PageLoadingActivityIndicator'; // Adjust the import path as needed

describe('PageLoadingActivityIndicator Component', () => {
  const mockTheme = {
    ...DefaultTheme,
    colors: {
      ...DefaultTheme.colors,
      background: '#ffffff',
      primary: '#6200ee',
    },
  };

  const wrapper = ({children}: any) => <ThemeProvider theme={mockTheme}>{children}</ThemeProvider>;

  it('renders correctly', () => {
    const {toJSON} = render(<PageLoadingActivityIndicator />, {wrapper});
    expect(toJSON()).toMatchSnapshot();
  });

  it('applies correct styles based on theme', () => {
    const {getByTestId} = render(<PageLoadingActivityIndicator />, {wrapper});

    const container = getByTestId('activity-indicator-container');
    const activityIndicator = getByTestId('activity-indicator');

    const containerStyle = container.props.style[1];
    const activityIndicatorProps = activityIndicator.props;

    // Check container background color
    expect(containerStyle.backgroundColor).toBe(mockTheme.colors.background);
    // Check ActivityIndicator color
    expect(activityIndicatorProps.color).toBe(mockTheme.colors.primary);
  });

  it('renders an ActivityIndicator with large size', () => {
    const {getByTestId} = render(<PageLoadingActivityIndicator />, {wrapper});

    const activityIndicator = getByTestId('activity-indicator');
    const activityIndicatorProps = activityIndicator.props;

    // Check ActivityIndicator size
    expect(activityIndicatorProps.size).toBe('large');
  });
});
