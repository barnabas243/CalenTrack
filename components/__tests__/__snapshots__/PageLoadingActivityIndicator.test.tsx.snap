// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoadingActivityIndicator Component renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      {
        "backgroundColor": "#ffffff",
      },
    ]
  }
  testID="activity-indicator-container"
>
  <ActivityIndicator
    color="#6200ee"
    size="large"
    testID="activity-indicator"
  />
</View>
`;
